### ISSUE TITLE
[HOLD for payment 2024-06-28] [$250] mWeb - Chat - Uploading an image while offline briefly shows an offline message in the preview

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#42657]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.76.1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4575560
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition: Go offline
1. Go to https://staging.new.expensify.com/home
2. Tap on a report
3. Tap plus icon - upload an attachment image

## Expected Result:
The image preview should be displayed even when uploading an attachment while offline.

## Actual Result:
Uploading an image while offline briefly shows an offline message in the preview

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/837b86ee-7cb5-4e45-b8d3-df723ca0f642

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017a93bd4c637b4a77</li>
        <li>Upwork Job ID: 1798155121003208704</li>
        <li>Last Price Increase: 2024-06-05</li>
<li>Automatic offers: </li>
<ul>
<li>allgandalf | Reviewer | 102625969</li>
<li>Krishna2323 | Contributor | 102625971</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @anmurali</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb

**Comment by kbecciv** [comment]
@anmurali FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by ashuvssut** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
While offline, when we add an local file attachment, the `<AttachmentOfflineIndicator />` shows up

### What is the root cause of that problem?
<AttachmentOfflineIndicator /> cannot distinguish whether the uploaded file is a local file or not, and it displays in both cases.

### What changes do you think we should make in order to solve the problem?
[This `<AttachmentOfflineIndicator /> ` component](https://github.com/Expensify/App/blob/cb692c5a1c8360db4309c2eb8212e6c636d8b851/src/components/AttachmentOfflineIndicator.tsx#L17-L18) should accept an `uri` prop. Based on the `uri`, it will judge if it has to show the offline indicator or not

https://github.com/Expensify/App/blob/cb692c5a1c8360db4309c2eb8212e6c636d8b851/src/components/AttachmentOfflineIndicator.tsx#L33-L35

```diff
+ const isLocalFile = uri.startsWith('blob:') || uri.startsWith('file:');
+ if (isLocalFile || !isOffline || onCacheDelay) {
- if (!isOffline || onCacheDelay) { 
```

We have 5 different components who use this `<AttachmentOfflineIndicator /> ` component. I have verified that each of those components can readily pass this new `uri` prop to  `<AttachmentOfflineIndicator /> `

### What alternative solutions did you explore? (Optional)

And easier solution could be to just change the rendering conditionals in [ImageView component](https://github.com/Expensify/App/blob/cb692c5a1c8360db4309c2eb8212e6c636d8b851/src/components/ImageView/index.tsx#L22)

That is, 
1. first create the conditional:-
	```tsx
	    const isLocalFile = url.startsWith('file:') || url.startsWith('blob:');
	    const showOfflineIndicator = isLoading && !isLocalFile;
	```
2. Apply the conditional

	```diff
	-        {isLoading && <AttachmentOfflineIndicator />}
	+       {showOfflineIndicator && <AttachmentOfflineIndicator />}
	+  </View>
	```

**Comment by arosiclair** [comment]
Thanks for the breakdown @allgandalf 

> But @dominictb also found out one more bug in the same flow that we do not show the loading indicator initially when the user uploads attachment when offline

This doesn't really seem like much of a bug to me just a very minor quirk. Let's ignore this. @Krishna2323's proposal looks pretty straight forward so let's go with that.

**Comment by arosiclair** [comment]
@dominictb Krishna posted a better proposal earlier than yours. Your proposal included a fix for another bug which I deemed not important. This is why I hired Krishna. Whether or not the bug you reported ends up getting fixed here or anywhere is not important. You do not "own" that bug and we never posted a job to fix it.

I'd appreciate it if you would let this go. The repeated back in forth in these comments is not really doing you any favors and this energy would be better spent on another proposal instead. Thank you 🙏
